siman/core/structure.py

```diff
@@ -274,12 +274,30 @@ def exchange_axes(self, i1_r, i2_r):
 
         st.rprimd[i1_r] = r[i2_r]
         st.rprimd[i2_r] = r[i1_r]
-
+        
         st.update_xred()
 
         return st
 
+    def exchange_axes_with_atoms(self, i1_r, i2_r):
+        """
+        
```

Comment:
Add description, the name of method should be refined, lets discuss

Reply:
exchange_axes_with_atoms_rotation?